openbas-api/src/main/java/io/openbas/rest/payload/PayloadUtils.java

```diff
@@ -146,39 +158,51 @@ public <T extends Payload> void duplicateCommonProperties(
     duplicate.setCollector(null);
     duplicate.setSource(Payload.PAYLOAD_SOURCE.MANUAL);
     duplicate.setStatus(Payload.PAYLOAD_STATUS.UNVERIFIED);
-    outputParserUtils.copyOutputParsers(origin.getOutputParsers(), duplicate, false);
+    outputParserService.copyOutputParsersFromEntity(origin.getOutputParsers(), duplicate);
 
     if (eeService.isLicenseActive(licenseCacheManager.getEnterpriseEditionInfo())) {
       detectionRemediationUtils.copy(origin.getDetectionRemediations(), duplicate, false);
     }
   }
 
-  public Payload copyProperties(Object payloadInput, Payload target, boolean copyId) {
+  public Payload copyProperties(PayloadCreateInput payloadInput, Payload target) {
+    if (payloadInput == null) {
+      throw new IllegalArgumentException("Input payload cannot be null");
+    }
+    BeanUtils.copyProperties(
+        payloadInput, target, "outputParsers", "tags", "attackPatterns", "detectionRemediations");
+
+    outputParserService.copyOutputParsersFromInput(payloadInput.getOutputParsers(), target);
+    detectionRemediationUtils.copy(payloadInput.getDetectionRemediations(), target, false);
+    return target;
+  }
+
+  public Payload copyProperties(PayloadUpdateInput payloadInput, Payload target) {
     if (payloadInput == null) {
       throw new IllegalArgumentException("Input payload cannot be null");
     }
 
     BeanUtils.copyProperties(
         payloadInput, target, "outputParsers", "tags", "attackPatterns", "detectionRemediations");
 
-    if (payloadInput instanceof PayloadCreateInput) {
-      outputParserUtils.copyOutputParsers(
-          ((PayloadCreateInput) payloadInput).getOutputParsers(), target, copyId);
-      detectionRemediationUtils.copy(
-          ((PayloadCreateInput) payloadInput).getDetectionRemediations(), target, copyId);
-    } else if (payloadInput instanceof PayloadUpdateInput) {
-      outputParserUtils.copyOutputParsers(
-          ((PayloadUpdateInput) payloadInput).getOutputParsers(), target, copyId);
-      detectionRemediationUtils.copy(
-          ((PayloadUpdateInput) payloadInput).getDetectionRemediations(), target, copyId);
-    } else if (payloadInput instanceof PayloadUpsertInput) {
-      outputParserUtils.copyOutputParsers(
-          ((PayloadUpsertInput) payloadInput).getOutputParsers(), target, copyId);
-      detectionRemediationUtils.copy(
-          ((PayloadUpsertInput) payloadInput).getDetectionRemediations(), target, copyId);
-    } else {
-      throw new IllegalArgumentException("Unsupported payload input type");
+    outputParserService.copyOutputParsersFromInput((payloadInput).getOutputParsers(), target);
+    detectionRemediationUtils.copy((payloadInput).getDetectionRemediations(), target, true);
+    return target;
+  }
+
+  public Payload copyProperties(
```

Comment:
Could we try and merge all 3 copyProperties methods into 1 using generics ? like public <U extends PayloadInput> Payload copyProperties(U payloadInput,
      Payload target,
      boolean copyId) ?